links/src/main/java/juja/microservices/links/repository/impl/LinksRepositoryImpl.java

```diff
@@ -1,18 +1,61 @@
 package juja.microservices.links.repository.impl;
 
 import juja.microservices.links.model.Link;
+import juja.microservices.links.model.SaveLinkRequest;
 import juja.microservices.links.repository.LinksRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.stereotype.Repository;
 
+import javax.inject.Inject;
 import java.util.Set;
 
-/**
- * @author Ivan Shapovalov
- */
+import static org.springframework.data.mongodb.core.query.Criteria.where;
+import static org.springframework.data.mongodb.core.query.Query.query;
+
+@Slf4j
+@Repository
 public class LinksRepositoryImpl implements LinksRepository {
+    @Inject
+    private MongoTemplate mongoTemplate;
+
+    @Value("${spring.data.mongodb.collection}")
+    private String mongoCollectionName;
 
     @Override
-    public Link saveLink() {
-        return null;
+    public Link saveLink(SaveLinkRequest request) {
+        Link link;
+        String URL = request.getURL();
```

Comment:
Strange logic. I think more than extra and dublicate.
String URL = request.getURL(); is pointless.
We can use 
Link currentLink=getLinkByURL(String URL). That's all.
If currentLink is null - save new Link. Otherwise - return currentLink.

Reply:
done